### PR TITLE
Break the GreyVest Modal/Popover API

### LINKED ISSUE #317: GreyVest - optional lens API
For v2, let's make modal and popover _optionally_ support lenses instead of requiring it.

Specifically:
`isOpen` becomes the bool (value), `toggle` (name up to you) becomes the setter function
`open` becomes the lens (which is just lensified).

### PR DESCRIPTION
**Pull request by michsa**
Closes #317 

this would have been a great spot for some kind of pun about `open`/`isOpen`

### CODE REVIEW

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>The changelog has not been updated. Please update the changelog.</td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>The version was not updated. Please update the version.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>The README has not been updated. Please update the README.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Could not find any browser results.</td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against ab4a7e6643b689176122e51e7f219147b44d64d4
</p>

File: src/greyVest/utils.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { lensify } from '../utils/react'
+import F from 'futil-js'
+
+let binding = (...lens) => ({
```
Comment (daedalus28) [inline src/greyVest/utils.js]: we should call this `open` or `openBinding` - a `binding` (as we've defined it) is a function that takes a lens and returns an object

File: src/greyVest/utils.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+  isOpen: F.view(...lens),
+  onClose: F.off(...lens),
+})
+export let bindLens = lensify(binding, 'open')
```
Comment (daedalus28) [inline src/greyVest/utils.js]: we should be consistent with material and call this `openify` (or rename `openify` in material)

File: src/utils/react.js [hunk 3]
```diff
@@ -14,17 +15,8 @@ export let wrapDisplayName = (name, Component) => Wrapped => {
...
-// ({a} -> {b}) -> {a} -> {a, b}
-let expandObject = _.curry((transform, obj) => ({ ...obj, ...transform(obj) }))
-
-// (a -> {b}, k) -> Component<{k: a}> -> Component<{b}>
-export let lensify = _.flow(
-  getWith,
-  expandObject,
+// (k, a -> {b}) -> Component<{k: a}> -> Component<{b}>
+export let expandProp = _.flow(
+  expandObjectBy,
```
Comment (daedalus28) [inline src/utils/react.js]: @stellarhoof FYI, you'll like this much more :)

File: src/themes/material/index.js [hunk 4]
```diff
@@ -30,7 +31,7 @@ export default {
   Icon,
   DropdownItem: MenuItem,
   PickerItem: MenuItem,
-  Modal: openify(Dialog),
+  Modal: expandProp('open', openBinding)(Dialog),
```
Comment (daedalus28) [inline src/themes/material/index.js]: @stellarhoof ☝️  👍 😄

Comment (stellarhoof) [reply src/themes/material/index.js]: Thank god. Phew.

File: src/utils/react.js (hunk 3 above)
Comment (stellarhoof) [reply src/utils/react.js]: This reads better for sure. I still don't like `getWith` though

**Comment by stellarhoof** [comment]
I gotta say I like `expandProp('open', openBinding)` better than `openify`. It's more explicit with little code. Nice :+1:

**Pull request merged**